examples/rust-generate-crate/index.ts

```diff
@@ -0,0 +1,69 @@
+import { RustFileGenerator, RustRenderCompleteModelOptions, RUST_COMMON_PRESET, defaultRustRenderCompleteModelOptions, RustPackageFeatures } from '../../src/generators';
+import * as path from 'path';
+
+const doc = {
+  $id: '_address',
+  type: 'object',
+  properties: {
+    street_name: { type: 'string' },
+    city: { type: 'string', description: 'City description' },
+    state: { type: 'string' },
+    house_number: { type: 'number' },
+    marriage: { type: 'boolean', description: 'Status if marriage live in given house' },
+    members: { oneOf: [{ type: 'string' }, { type: 'number' }, { type: 'boolean' }], },
+    tuple_type: { type: 'array', items: [{ type: 'string' }, { type: 'number' }], additionalItems: false },
+    array_type: { type: 'array', items: { type: 'string' }, additionalItems: false },
+    enum_type: {
+      enum: ['Texas', 'Alabama', 'California'],
+      default: 'California'
+    }
+  },
+  required: ['street_name', 'city', 'state', 'house_number', 'array_type'],
+  additionalProperties: {
+    type: 'string'
+  },
+};
+
+export async function generate(): Promise<void> {
+  // initialize the generator from a preset
+  const generator = new RustFileGenerator({
+    presets: [
+      {
+        preset: RUST_COMMON_PRESET, options: {
+          implementNew: true,
+          implementDefault: true
+        }
+      }
+    ]
+  });
+  // Generated files will be written to output/ directory
+  const outDir = path.join(__dirname, 'output');
+  // Parse the schema document
+  const inputModel = await generator.process(doc);
+
+  // Run the file generator with options
+  const models = await generator.generateToFiles(inputModel, outDir, {
+    ...defaultRustRenderCompleteModelOptions,
+    supportFiles: true, // generate Cargo.toml and lib.rs
+    package: {
+      packageName: 'asyncapi-rs-example',
+      packageVersion: '1.0.0',
+      // set authors, homepage, repository, and license
+      authors: ['AsyncAPI Rust Champions'],
+      homepage: 'https://www.asyncapi.com/tools/modelina',
+      repository: 'https://github.com/asyncapi/modelina',
+      license: 'Apache-2.0',
+      description: 'Rust models generated by AsyncAPI Modelina',
+      // support 2018 editions and up
+      edition: '2018',
+      // enable serde_json
+      packageFeatures: [RustPackageFeatures.json] as RustPackageFeatures[]
+    }
```

Comment:
I can cut the user-facing API to avoid scope creep.

We do need a `Cargo.toml` manifest and `lib.rs` module for black-box testing. Otherwise, compilation commands like `cargo build` and sanity checks like `cargo clippy` will fail. Would it make more sense to include this functionality in a test helper instead?

Reply:
> We do need a Cargo.toml manifest and lib.rs module for black-box testing. Otherwise, compilation commands like cargo build and sanity checks like cargo clippy will fail. Would it make more sense to include this functionality in a test helper instead?

I think only you can answer that 😄 If you see it as a benefit that the user can generate the support files, then I am all good with keeping it 🙂